logrotate.c

```diff
@@ -510,8 +510,13 @@ static int createOutputFile(char *fileName, int flags, struct stat *sb,
     if (fchmod(fd, (S_IRUSR | S_IWUSR) & sb->st_mode)) {
 	message(MESS_ERROR, "error setting mode of %s: %s\n",
 		fileName, strerror(errno));
-	close(fd);
-	return -1;
+
+		if (geteuid() == 0) {
+			/* If run as non-root, we may not be able to duplicate
+			the ownership of the file. Warn but don't fail. */
```

Comment:
Looking once again, the condition is not accurate on Linux anyway.  A process running with the ```CAP_CHOWN``` capability can change the ownership even if ```geteuid() != 0```.